File: kafka-producer/src/main/java/KafkaProducerSample.java
import org.apache.kafka.clients.producer.KafkaProducer;
import org.apache.kafka.clients.producer.ProducerRecord;

import java.util.Properties;

/**
 * Created by jazz on 2017/02/09.
 *
 * Reference:
 * [1] https://github.com/apache/kafka/blob/trunk/examples/src/main/java/kafka/examples/Producer.java
 * [2] https://kafka.apache.org/0100/javadoc/index.html?org/apache/kafka/clients/producer/KafkaProducer.html
 * https://github.com/bigdatabiginsight/kafka/blob/master/kafka-producer2/src/main/java/com/learningstorm/kafka/WordsProducer.java
 */
public class KafkaProducerSample {
    public static void main(String[] args)
    {
        Properties props = new Properties();
        props.put("bootstrap.servers", "localhost:9092");
        props.put("key.serializer", "org.apache.kafka.common.serialization.StringSerializer");
        props.put("value.serializer", "org.apache.kafka.common.serialization.StringSerializer");
        KafkaProducer<String,String> producer = new KafkaProducer<>(props);
        for (int i = 1; i <= 50_000; i++) {
            producer.send(new ProducerRecord<String, String>("test", Integer.toString(i), Integer.toString(i)));
            System.out.println("Sending to topic 'test' with key '" + i + "' and payload '" + i + "'");
        }
        producer.flush();
        producer.close();
    }
}

File: js/kafkajs/KafkaProducerSample.js
const { Kafka } = require('kafkajs');

const kafka = new Kafka({
    clientId: 'my-app',
    brokers: ['localhost:9092']
});

const producer = kafka.producer();

const run = async () => {
    await producer.connect();
    for (let i = 1; i <= 50000; i++) {
        await producer.send({
            topic: 'test',
            messages: [
                { key: i.toString(), value: i.toString() },
            ],
        });
        console.log(`Sending to topic 'test' with key '${i}' and payload '${i}'`);
    }
    await producer.disconnect();
};

run().catch(console.error);
